app/common/collections/runner.py

```diff
@@ -0,0 +1,179 @@
+# todo: propose moving common/helper/collections.py to common/collections/submission.py
+from typing import TYPE_CHECKING, Callable, ClassVar, Optional
+from uuid import UUID
+
+from app.common.collections.forms import CheckYourAnswersForm, build_question_form
+from app.common.data.types import FormRunnerState, SubmissionStatusEnum
+from app.common.forms import GenericSubmitForm
+from app.common.helpers.collections import SubmissionHelper
+from app.extensions import notification_service
+
+if TYPE_CHECKING:
+    from app.common.collections.forms import DynamicQuestionForm
+    from app.common.data.models import Form, Question
+    from app.common.data.models_user import User
+
+runner_url_map = dict[
+    FormRunnerState, Callable[["FormRunner", Optional["Question"], Optional["Form"], Optional[FormRunnerState]], str]
+]
+
+
+class FormRunner:
```

Comment:
This whole encapsulation feels really clean and the change it makes to what goes on in the views is night and day - very nice idea 🙇